### ISSUE TITLE
ScanNet dataset 

### ISSUE THREAD

**Issue by BenChenCh** [Yang7879/3D-BoNet#6]
@Yang7879 Could you  share the code to make dataset with  scanNet (v2) or describe how to do it? Thanks

**Comment by Yang7879** [comment]
@BenChenCh It's pretty easy. 
(1) Use the ScanNet official code (https://github.com/ScanNet/ScanNet/tree/master/BenchmarkScripts/3d_helpers) to extract raw point clouds and the sem+ins ground truth. 
(2) To divide the large point clouds into 1mx1m blocks. Refer to PointNet code (https://github.com/charlesq34/pointnet/blob/master/sem_seg/indoor3d_util.py)
(3) If you need to use SCN to train the point semantic segmentation, refer to code (https://github.com/facebookresearch/SparseConvNet/tree/master/examples/ScanNet).

Note, just remove the "psemce_loss" from the "end_2_end_loss" , if you don't use the pointnet++ based semantic segmentation branch.

**Comment by 519830100** [comment]
Hi @Yang7879  Does the training process input all the points from 40 classes or just fewer points from 18 classes or 20 classes (including wall and ground). Some papers input points that belong to 18 classes, would this cause a problem during the testing process, which has to input all points?

**Comment by Yang7879** [comment]
Hi @519830100 In my case, the sparseCov was trained on 20 cls (same as their settings) for semantic seg;  for instance segmentation, I tried either 20 cls or 40 cls for training, the results are quite similar. This is sensible cuz our 3D-BoNet doesn't require the semantic label for training.

**Comment by tonghe90** [comment]
Hi @Yang7879, Thanks for your reply. what I mean is: Do you remove the other points from the input? Or you input all the points and ignore them in the loss function?

**Comment by bonbonjour** [comment]
@BenChenCh，hi，I am interested in trainning with scannet database, too. Have you successfully generate raw point clouds and the sem+ins ground truth? can you please tell exactly steps to  extract raw point clouds and the sem+ins ground truth. thanks

**Comment by fxy2012** [comment]
@Yang7879 , hi, I can't understand the meaning of the sem+ins ground truth.  Does it mean `export_semantic_label_grid_for_evaluation`? What is the meaning of  "ins" ?

**Comment by Violetit** [comment]
@Yang7879 @BenChenCh @bonbonjour 
Hi, I have some trouble with extracting the semantic and instance labels on ScanNet_v2.
Cloud you please tell exactly steps to extract raw point clouds and the sem+ins ground truth?
Hope to hear from you soon！
